apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -1287,6 +1323,11 @@ def _create_real_external_network(self, context, network):
         if not net_info:
             return
 
+        # if there is a router_type (like ASR) then we don't need to create
+        # this L3 out, NatEPG, contract, ...etc in APIC
+        if self._is_asr_router_type(net_info):
+            return
```

Comment:
I see.....so I think I should also create the NAT VRF, EPG and BD if its not pre-existing mode. will do that then. Let me know if something is still missing then.